Willster419/RelhaxModpack
# merging latest changes from developer into the beta application

Ok so i know this is a lot. But the last few months have been, uh, interesting for me. So there's my non-excuse for having a large roll-up of changes.

- Atlas building was improved
  -  Instead of using the `GetPixel` and `SetPixel` (See https://github.com/Willster419/RelhaxModpack/commit/042cb8d46fc0f46992fda992eb75d1a94ab82f36 for sample)
  -  Lock the entire atlas bitmap and marshal it to a byte array
  -  Calculate out to the start of the rectangle of the image to replace
  -  Replace that line of the image stride with the new data
  -  Repeat for each stride line until the bottom of the image
  -  This method is exponentially faster (time reduced by 96%) and keeps the bit-for-bit accuracy
- Added a check to ensure the user has the .NET Framework version 4.8 or later installed.
  -  I don't see us moving from this version for a while, while we still support windows 7.
- Re-worked the shortcut code to use one imported COM reference. Allows us to remove the COM reference for `IWshRuntimeLibrary`
- Over-ridded the `BeforeBuild` target with nuget dll copying (it was nothing before, no `msbuild` target code lost
  -  Creates an `Item` in an `ItemGroup` to get all `Reference` objects with a `HintPath`. Those seem to be only nuget packages
  -  Gets the paths to all the nuget `dll`s
  -  Copies the `dll`s into the project's `DLL` folder, overriding the embedded versions.
  -  This essentially turns into automation to auto-update the embedded `dll`s, when we update nuget packages
- Added `ZipImageLibs` target to inject after `BeforeBuild` target
  -  Gets the file paths to the 32 and 64bit `nvtt.dll` and `freeimage.dll` in `TeximpNet`
  -  Creates a zip file of each int othe `DLL` folder, again overriding the embedded versions
- The `msbuild` work also meant that we updated almost all of our nuget packages
- Renamed `DatabaseUpdater` to `ModpackToolbox`
  -  Added feature to export translations to a tab-delimited csv file, with text escaped
- Removed setting to save database as legacy in the editor. No one uses it
- Added several utility methods to the `Translations` class
- Several comment and logfile writing improvements

TO TEST:
- Run the unit test (make sure it's in 64bit mode) and ensure everything passes
- Run an install if you want and make sure there's no errors or exceptions in the log
  -  Then a replay would show the results

Would also appreciate a review of the code itself, especially the atlas part.
